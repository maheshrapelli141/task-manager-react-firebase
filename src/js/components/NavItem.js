import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const NavItem = ({ path, label }) => (
  <li className="nav-item">
    <NavLink to={path} activeClassName="active" className="nav-link">
      {label}
    </NavLink>
  </li>
);

NavItem.propTypes = {
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default NavItem;
