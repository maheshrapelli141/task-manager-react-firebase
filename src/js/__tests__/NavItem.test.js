import React from 'react';
import { shallow } from 'enzyme';

import NavItem from '../components/NavItem';

const wrapper = shallow(<NavItem path="/test-path" label="Test Label" />);

describe('NavItem component', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
