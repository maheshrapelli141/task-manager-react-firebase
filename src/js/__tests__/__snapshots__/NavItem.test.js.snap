// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem component should render correctly 1`] = `
<li
  className="nav-item"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link"
    to="/test-path"
  >
    Test Label
  </NavLink>
</li>
`;
